plugins/techdocs-react/src/context.test.tsx

```diff
@@ -55,72 +56,137 @@ const techdocsApiMock = {
+        </TestApiProvider>
+      </ThemeProvider>
     );
 
-    expect(result.current.title).toBe('');
+    it('should set title', async () => {
+      const { result, waitForNextUpdate } = renderHook(
+        () => useTechDocsReaderPage(),
+        { wrapper },
+      );
 
-    act(() => result.current.setTitle('test site title'));
+      expect(result.current.title).toBe('');
 
-    await waitForNextUpdate();
+      act(() => result.current.setTitle('test site title'));
 
-    expect(result.current.title).toBe('test site title');
-  });
+      await waitForNextUpdate();
 
-  it('should set subtitle', async () => {
-    const { result, waitForNextUpdate } = renderHook(
-      () => useTechDocsReaderPage(),
-      { wrapper },
-    );
+      expect(result.current.title).toBe('test site title');
+    });
 
-    expect(result.current.subtitle).toBe('');
+    it('should set subtitle', async () => {
+      const { result, waitForNextUpdate } = renderHook(
+        () => useTechDocsReaderPage(),
+        { wrapper },
+      );
 
-    act(() => result.current.setSubtitle('test site subtitle'));
+      expect(result.current.subtitle).toBe('');
 
-    await waitForNextUpdate();
+      act(() => result.current.setSubtitle('test site subtitle'));
 
-    expect(result.current.subtitle).toBe('test site subtitle');
-  });
+      await waitForNextUpdate();
 
-  it('should set shadow root', async () => {
-    const { result, waitForNextUpdate } = renderHook(
-      () => useTechDocsReaderPage(),
-      { wrapper },
-    );
+      expect(result.current.subtitle).toBe('test site subtitle');
+    });
+
+    it('should set shadow root', async () => {
+      const { result, waitForNextUpdate } = renderHook(
+        () => useTechDocsReaderPage(),
+        { wrapper },
+      );
 
-    // mock shadowroot
-    const shadowRoot = mockShadowRoot();
+      // mock shadowroot
+      const shadowRoot = mockShadowRoot();
 
-    act(() => result.current.setShadowRoot(shadowRoot));
+      act(() => result.current.setShadowRoot(shadowRoot));
 
-    await waitForNextUpdate();
+      await waitForNextUpdate();
 
-    expect(result.current.shadowRoot?.innerHTML).toBe(
-      '<h1>Shadow DOM Mock</h1>',
+      expect(result.current.shadowRoot?.innerHTML).toBe(
+        '<h1>Shadow DOM Mock</h1>',
+      );
+    });
+
+    it('should set entityRef as lowercase', async () => {
+      const lowercaseEntityRef = {
+        kind: mockEntityMetadata.kind.toLocaleLowerCase(),
+        name: mockEntityMetadata.metadata.name.toLocaleLowerCase(),
+        namespace: mockEntityMetadata.metadata.namespace?.toLocaleLowerCase(),
+      };
+      const { result } = renderHook(() => useTechDocsReaderPage(), { wrapper });
+
+      expect(result.current.entityRef).toStrictEqual(lowercaseEntityRef);
+    });
+  });
+
+  describe('when legacyUseCaseSensitiveTripletPaths is true', () => {
+    const configApiMock = {
+      getOptionalBoolean: jest.fn().mockReturnValue(true),
```

Comment:
I suggest a small refactoring in this test to avoid defining "wrapper" twice:

Have only one "wrapper" definition in the root describe block and your config API mock as a global variable containing a getOptionalBoolean that returns undefined by default.

`const configApiMock = { getOptionalBoolean: jest.fn().mockReturnValue(undefined) };`

So here in this test you can change the return value once (`configApiMock.getOptionalBoolean.mockReturnValueOnce(true)`) before rendering the hook instead of defining a wrapper again.

Reply:
Thanks for the pointers, this should be fixed!